go/vt/vtgate/engine/set.go

```diff
@@ -472,11 +472,14 @@ func (svss *SysVarSetAware) Execute(ctx context.Context, vcursor VCursor, env *e
 		if err != nil {
 			return err
 		}
-		out, ok := isolationLevelToExecuteOptionIsolationLevel[strings.ToLower(str)]
+		upperStr := strings.ToUpper(str)
+		out, ok := isolationLevelToExecuteOptionIsolationLevel[upperStr]
 		if !ok {
 			return vterrors.NewErrorf(vtrpcpb.Code_INVALID_ARGUMENT, vterrors.WrongValueForVar, "Variable '%s' can't be set to the value of '%s'", svss.Name, str)
 		}
-		vcursor.Session().SetTransactionIsolation(querypb.ExecuteOptions_TransactionIsolation(out))
+		vcursor.Session().SetTransactionIsolation(out)
+		vcursor.Session().SetSysVar(sysvars.TxIsolation.Name, fmt.Sprintf("'%s'", upperStr))
```

Comment:
I'm not really understanding why you write it twice to the session under different names. is it really needed?

Reply:
tx_isolation is the old system variable, transaction_isolation was introduced in 5.7.20. 
In 8.0, tx_isolation is removed.
just to support the case where the client set tx_isolation and selects transaction_isolation, this is done.

Reply:
ah, good to know. maybe we should add a comment saying this?